Description: Don't block SIGSEGV at more places.
 This is used internally by qemu to detect writing to pages
 containing code, it must not be disabled even when the guest
 disables these.
Author: Michael Matz <matz@suse.de>
Origin: https://github.com/susematz/qemu/commit/f1542ae9fe10d5a241fc2624ecaef5f0948e3472
Last-Update: 2014-02026

Index: qemu/linux-user/signal.c
===================================================================
--- qemu.orig/linux-user/signal.c	2014-02-27 16:37:49.355641961 -0600
+++ qemu/linux-user/signal.c	2014-02-27 16:37:49.351641961 -0600
@@ -1213,7 +1213,7 @@ static int target_restore_sigframe(CPUAR
     uint64_t pstate;
 
     target_to_host_sigset(&set, &sf->uc.tuc_sigmask);
-    sigprocmask(SIG_SETMASK, &set, NULL);
+    do_sigprocmask(SIG_SETMASK, &set, NULL);
 
     for (i = 0; i < 31; i++) {
         __get_user(env->xregs[i], &sf->uc.tuc_mcontext.regs[i]);
@@ -5753,6 +5753,7 @@ void process_pending_signals(CPUArchStat
             sigaddset(&set, target_to_host_signal(sig));
 
         /* block signals in the handler using Linux */
+        sigdelset(&set, SIGSEGV);
         sigprocmask(SIG_BLOCK, &set, &old_set);
         /* save the previous blocked signal state to restore it at the
            end of the signal execution (see do_sigreturn) */
